pkg/machines/libvirt-dbus.js

```diff
@@ -379,11 +379,15 @@ LIBVIRT_DBUS_PROVIDER = {
                     });
         }
 
-        if (options.destroy) {
+        if (options.destroy)
             return destroy().then(undefine());
-        } else {
-            return undefine();
-        }
+        else
+            return undefine()
+                    .catch(ex => {
+                        // Transient domains get undefined after shut off
+                        if (!ex.message.includes("Domain not found"))
+                            return cockpit.reject(ex);
```

Comment:
Not sure right now, but are these really cockpit promises? Can we avoid introducing new ones, and use standard `Promise`s?

Reply:
> [this shows](https://209.132.184.41:8493/logs/pull-12177-20190714-101212-90a24a61-verify-fedora-30/log.html#221) a real regression: There's a duplicate `key=` and ambiguous selector in the test. Check the screenshot, myPoolTwo is actually being shown twice, once active and once inactive.

Right, I just saw this in another PR as well, see https://logs-https-cockpit.apps.ci.centos.org/logs/pull-12302-20190713-094444-8e78f21f-verify-fedora-30/log.html#235

I will try to fix it here.